server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -2160,27 +2169,35 @@ public boolean match(String toTest) {
     public static final class AffixKey implements Key {
         private final Pattern pattern;
         private final String prefix;
+        private final String fallbackPrefix;
         private final String suffix;
 
         private final String keyString;
 
-        AffixKey(String prefix) {
-            this(prefix, null);
-        }
-
-        AffixKey(String prefix, String suffix) {
+        AffixKey(String prefix, String suffix, String fallbackPrefix) {
```

Comment:
should we call deprecation logger when a fallback prefix was used?

Reply:
Added.

Reply:
I'm moving this and related changes to another PR because it would be a backwards incompatible change.